internal/services/springcloud/spring_cloud_gateway_route_config_resource.go

```diff
@@ -64,6 +64,16 @@ func resourceSpringCloudGatewayRouteConfig() *pluginsdk.Resource {
 				},
 			},
 
+			"protocol": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				Default:  string(appplatform.GatewayRouteConfigProtocolHTTP),
```

Comment:
this is an insecure default, what's the reasoning for making this `HTTP` rather than `HTTPS` (or Required?)

Reply:
The server-side's default value is `HTTP`, if we make it defaults to `HTTPS` or required field. It will introduce breaking change.

Reply:
Can we make this field Required for 4.0 then and also add a note in the docs warning users that this is insecure by default.

Reply:
Thanks @stephybun , I've updated the PR, please take another look.